src/aiohomeconnect/client.py

```diff
@@ -152,6 +189,11 @@ async def get_all_programs(
             f"/homeappliances/{ha_id}/programs",
             headers={"Accept-Language": accept_language},
         )
+        match response.status_code:
+            case 409:
+                raise Conflict.from_dict(response.json()["error"])
+            case _:
+                response.raise_for_status()
```

Comment:
We should have a library specific base response exception that we raise. We should not let httpx exceptions leak out.